tests/unittests/tests-fmt/tests-fmt.c

```diff
@@ -242,12 +242,36 @@ static void test_fmt_hex_bytes(void)
     TEST_ASSERT_EQUAL_INT(0xAF, val3[2]);
 }
 
+static void test_fmt_u16_hex(void)
+{
+    char out[8] = "zzzzzzz";
+    uint16_t val = 48879;
+
+    out[4] = '\0';
+
+    /* Check return count with null buffer input */
+    TEST_ASSERT_EQUAL_INT(4, fmt_u16_hex(NULL, val));
+    /* always four regardless of value size */
+    TEST_ASSERT_EQUAL_INT(4, fmt_u16_hex(NULL, 0));
+
+    TEST_ASSERT_EQUAL_INT(4, fmt_u16_hex(out, val));
+    TEST_ASSERT_EQUAL_STRING("BEEF", (char *) out);
+
+    /* check that the buffer was not overflowed */
+    TEST_ASSERT_EQUAL_STRING("zz", &out[5]);
+}
+
 static void test_fmt_u32_hex(void)
 {
     char out[12] = "zzzzzzzzzzz";
     uint32_t val = 3735928559;
 
     out[8] = '\0';
+    /* Check return count with null buffer input */
+    TEST_ASSERT_EQUAL_INT(8, fmt_u32_hex(NULL, val));
+    /* always four regardless of value size */
```

Comment:
```suggestion
    /* always eight regardless of value size */
```

Or just make the comment more generic